### ISSUE TITLE
Add customization for size of history and auto-complete boxes to the "Appearance" pane of the settings dialog.

### ISSUE THREAD

**Issue by lcrespom** [warpdotdev/warp#2583]
### Discord username (optional)

Luis#2191

### Describe the solution you'd like?

```text
Add more customization options to the "Appearance" pane of the settings dialog.

In particular, support customization for the number of lines in the history and auto-complete boxes.

The history and auto-complete boxes are quite small, allowing for only 6 elements, which gives a 
very narrow view and forces the user to scroll until they find the file/command they were looking for.

Larger boxes would help the user find a command or file much faster at a glance.
```

### Is your feature request related to a problem? Please describe.

```text
A larger file auto-complete box, showing more files and with more details about the currently
selected file (e.g. size, date, attributes...), would provide a full file navigator to be used
whenever a file or path is required.

With their current small sizes, I rarely use the command history or file auto-complete features,
as they force me to scroll one item at a time because the one I am looking for is out of sight.
```

### How important is this feature to you?

4

### Warp Internal (ignore) - linear-label:770f6576-d6c0-4e4f-a259-fc64b5156087

None

**Comment by dannyneira** [comment]
Thanks for your feedback and suggestion. :+1: this request as it helps us gauge interest.

**Comment by Akirathan** [comment]
I believe this is a duplicate of #3980, and probably related to #4984.

**Comment by Akirathan** [comment]
For more context, I am attaching few screenshots to compare Warp (v0.2024.11.12.08.02.stable_02) with my gnome-terminal with fzf:

---------------------------------------
History in gnome-terminal:
- Number of history items is OK - definitely better than just 6
- History items take the whole width of the terminal - better than the behavior on Warp
![Pasted image (3)](https://github.com/user-attachments/assets/33b07614-e8bc-4c9e-9d20-bc631aa64c63)

---------------------------------------
History in Warp (pressing CTRL+R and then typing something):
- Number of history items is acceptable.
- Width of the windows is too small, I can hardly differentiate the items.
![Pasted image (2)](https://github.com/user-attachments/assets/8e8a9ebd-6122-408e-8e8d-07a2b25a6aea)

---------------------------------------
History in Warp (pressing Up arrow):
- Number of history items is too small - just 4 items.
- Width is OK here.
![Pasted image](https://github.com/user-attachments/assets/3a55d7c0-3a02-4192-afec-eb43d3d1de8d)

**Comment by zachbai** [comment]
@Akirathan Thanks for raising. At least for the width issue with CTRL-R history, you can drag the edge of the modal to increase its width, which will be persisted and used the next time you access it

We can look into increasing the # of items shown in up-arrow history, should be a trivial fix.

**Comment by lcrespom** [comment]
Yes, and the same for the tab-completion, it's also too small to be useful.

**Comment by rcy007** [comment]
The small list of sub-directories is also a bit misleading as there's no indication ( down arrow ) to tell that there are actually more sub-directories than 6. I just realized there's a scroll bar on hover, but ideally you don't use mouse while going through directories.

**Comment by jerryzhou196** [comment]
+1 on this issue - any updates @zachbai?

**Comment by dimitarvdimitrov** [comment]
currently the history stored in sqlite is limited to 10000 items. It would be nice to have this configurable. I would prefer to keep a longer history at the cost of slower lookups

**Comment by acarl005** [comment]
This can now be resized by grabbing the edge with the mouse and dragging.

**Comment by dimitarvdimitrov** [comment]
@acarl005 this still don't resolve the item limit in history
